python/meep.i

```diff
@@ -0,0 +1,99 @@
+/* Copyright (C) 2005-2017 Massachusetts Institute of Technology  
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software Foundation,
+ *  Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+%module(directors="1") meep
+
+%{
+#define SWIG_FILE_WITH_INIT
+#include "meep/vec.hpp"
+#include "meep.hpp"
+using namespace meep;
+%}
+
+%include "numpy.i"
+
+%{
+#if NPY_API_VERSION < 0x00000007
+#define NPY_ARRAY_C_CONTIGUOUS NPY_C_CONTIGUOUS
+#define NPY_ARRAY_ALIGNED  NPY_ALIGNED
+#endif
+%}
+
+%init %{
+  import_array();
+%}
+
+// TODO: apply necessary numpy typemaps
+
+// Rename python builtins
```

Comment:
See also
```
%include "meep_renames.i"
%include "meep_enum_renames.i"
%include "meep_op_renames.i"
%include "meep_swig_bug_workaround.i"
```
from the libctl directory.  Can we re-use anything?

Couldn't all this `%rename` and `%feature` cruft be put into separate `.i` files?